probe-rs-tools/src/bin/probe-rs/util/cli.rs

```diff
@@ -478,18 +478,27 @@ pub async fn monitor(
     })
     .await;
 
-    let print_stack_trace = match &result {
+    match result {
         Ok(MonitorExitReason::Success | MonitorExitReason::SemihostingExit(Ok(_))) => {
             println!("Firmware exited successfully");
-            print_stack_trace // On success, we only print if the user asked for it.
+            // On success, we only print if the user asked for it.
+            if print_stack_trace {
+                display_stack_trace(session, path).await?;
+            }
+            Ok(())
```

Comment:
The PR makes sense, thank you for that, but could we reduce the logic duplication a bit? You could assign `(print_stack_trace, Ok(()))` and `(true, Err(anyhow!(...)))` from the match arms.

Reply:
Absolutely, if that's what you prefer. Changed